### PR TITLE
feat: Collaboration

### PR DESCRIPTION
**Pull request by rob-at-cortex**

### CODE REVIEW

**Comment by f-steel** [comment]
I can't seem to see this branch locally for some reason. Is it a private branch?

File: src/_components/Editor/Editor.tsx [hunk 1]
```diff
@@ -241,37 +421,52 @@ const Editor = React.memo(function EditorC({
...
+            } catch (err: any) {
+              errorRef.current = err.message;
+            } finally {
+              setIsLoading(false);
+            }
+          }}
+        >
+          {isLoading ? <CircularProgress size={24} /> : <SaveIcon />}
+        </Fab>
+        <Snackbar
```
Comment (f-steel) [inline src/_components/Editor/Editor.tsx:445]: removed the toast component in https://github.com/AirWalk-Digital/airview-next/pull/49 to be consistent with this :)

**Comment by f-steel** [comment]
```
        <div
            style={{
              marginTop: topBarHeight,
              paddingLeft: 0,
            }}
          >
            <EditorWrapper
              defaultContext={siteConfig.content[contentKey]}
              context={contentConfig}
              branches={branches}
            />
          </div>
```
this is the component that seems to be causing the cursor to appear at the wrong height. Somehow the Editor will need to compensate for the marginTop applied to the div that wraps around the editorWrapper

**Comment by f-steel** [comment]
Screenshot 2024-07-09 at 11 40 51
seems to be resolved with latest commit

**Comment by rob-at-cortex** [comment]
nice!

**Review by f-steel — APPROVED** [review]

**Pull request merged**